luaui/Include/blueprint_substitution/definitions.lua

```diff
@@ -0,0 +1,166 @@
+    DefCat("TORPEDO", {[SIDES.ARM]="armdl", [SIDES.CORE]="cordl", [SIDES.LEGION]="legdl"})
+    DefCat("ADV_TORPEDO", {[SIDES.ARM]="armatl", [SIDES.CORE]="coratl", [SIDES.LEGION]="legatl"})
+    DefCat("OFFSHORE_TORPEDO", {[SIDES.ARM]="armptl", [SIDES.CORE]="corptl", [SIDES.LEGION]="legptl"})
+    DefCat("ARTILLERY", {[SIDES.ARM]="armguard", [SIDES.CORE]="corpun", [SIDES.LEGION]="legcluster"})
+    DefCat("LONG_RANGE_PLASMA_CANNON", {[SIDES.ARM]="armbrtha", [SIDES.CORE]="corint", [SIDES.LEGION]="leglrpc"})
+    DefCat("RAPID_FIRE_LONG_RANGE_PLASMA_CANNON", {[SIDES.ARM]="armvulc", [SIDES.CORE]="corbuzz", [SIDES.LEGION]="legstarfall"})
+    DefCat("ANNIHILATOR", {[SIDES.ARM]="armanni", [SIDES.CORE]="cordoom", [SIDES.LEGION]="legbastion"})
+    DefCat("ADVANCED_PLASMA_ARTILLERY", {[SIDES.ARM]="armamb", [SIDES.CORE]="cortoast", [SIDES.LEGION]="legacluster"})
+    DefCat("DRAGONS_CLAW", {[SIDES.ARM]="armclaw", [SIDES.CORE]="cormaw", [SIDES.LEGION]="legdrag"})
+    DefCat("DRAGONS_TEETH", {[SIDES.ARM]="armdrag", [SIDES.CORE]="cordrag", [SIDES.LEGION]="legdrag"})
+    DefCat("ADVANCED_DRAGONS_TEETH", {[SIDES.ARM]="armfort", [SIDES.CORE]="corfort", [SIDES.LEGION]="legforti"})
+
+    -- Intel and special buildings
+    DefCat("RADAR", {[SIDES.ARM]="armrad", [SIDES.CORE]="corrad", [SIDES.LEGION]="legrad"})
+    DefCat("ADVANCED_RADAR", {[SIDES.ARM]="armarad", [SIDES.CORE]="corarad", [SIDES.LEGION]="legarad"})
+    DefCat("ADV_RADAR", {[SIDES.ARM]="armarad", [SIDES.CORE]="corarad", [SIDES.LEGION]="legarad"})
+    DefCat("JAMMER", {[SIDES.ARM]="armjamt", [SIDES.CORE]="corjamt", [SIDES.LEGION]="legjam"})
+    DefCat("ADVANCED_JAMMER", {[SIDES.ARM]="armveil", [SIDES.CORE]="corshroud", [SIDES.LEGION]="legajam"})
+    DefCat("SONAR", {[SIDES.ARM]="armsonar", [SIDES.CORE]="corsonar", [SIDES.LEGION]="legsonar"})
+    DefCat("ADV_SONAR", {[SIDES.ARM]="armason", [SIDES.CORE]="corason", [SIDES.LEGION]="legason"})
+    DefCat("CAMERA", {[SIDES.ARM]="armeyes", [SIDES.CORE]="coreyes", [SIDES.LEGION]="legeyes"})
+    DefCat("NUKE", {[SIDES.ARM]="armsilo", [SIDES.CORE]="corsilo", [SIDES.LEGION]="legsilo"})
+    DefCat("ANTINUKE", {[SIDES.ARM]="armamd", [SIDES.CORE]="corfmd", [SIDES.LEGION]="legabm"})
+    DefCat("JUNO", {[SIDES.ARM]="armjuno", [SIDES.CORE]="corjuno", [SIDES.LEGION]="legjuno"})
+    DefCat("NANO_TOWER", {[SIDES.ARM]="armnanotc", [SIDES.CORE]="cornanotc", [SIDES.LEGION]="legnanotc"})
+    DefCat("ADV_NANO_TOWER", {[SIDES.ARM]="armnanotct2", [SIDES.CORE]="cornanotct2", [SIDES.LEGION]="legnanotct2"})
+    DefCat("STEALTH_DETECTION", {[SIDES.ARM]="armrsd", [SIDES.CORE]="corrsd", [SIDES.LEGION]="legsd"})
+    DefCat("PINPOINTER", {[SIDES.ARM]="armtarg", [SIDES.CORE]="cortarg", [SIDES.LEGION]="legtarg"})
+    DefCat("SHIELD", {[SIDES.ARM]="armgate", [SIDES.CORE]="", [SIDES.LEGION]="legdeflector"})
+    DefCat("MEDIUM_RANGE_MISSILE", {[SIDES.ARM]="armemp", [SIDES.CORE]="cortron", [SIDES.LEGION]="legperdition"})
```

Comment:
Note: I could imagine a world where 
1) these categories are more complete
2) we discover other uses for them
3) they get moved out of our category-centric `definitions.lua` and onto the unit definitions themselves.